lib/utils/include/utils/graph/cow_ptr_t.h

```diff
@@ -90,21 +95,33 @@ struct cow_ptr_t {
     swap(lhs.ptr, rhs.ptr);
   }
 private:
-  void set_shared(shared_t ptr) {
-    this->ptr = variant<shared_t>(std::move(ptr));
+  void set_shared(shared_t ptr) const {
+    this->ptr = variant<
+    std::unique_ptr<T>,
+    std::shared_ptr<T const>
+  >(std::move(ptr));
   }
 
-  void set_unique(unique_t ptr) {
-    this->ptr = variant<unique_t>(std::move(ptr));
-  }
+ void set_unique(std::unique_ptr<T> ptr)  const{
+  this->ptr = variant<
+    std::unique_ptr<T>,
+    std::shared_ptr<T const>
+  >(std::move(ptr));
+}
 
-  std::unique_ptr<T> &get_unique() const {
-    return get<unique_t>(this->ptr);
+ std::unique_ptr<T> get_unique() const {
+  if (auto ptr = mpark::get_if<unique_t>(&this->ptr)) {
+    return std::move(*ptr);
   }
+  return nullptr;
+}
 
-  std::shared_ptr<T const> &get_shared() const {
-    return get<shared_t>(this->ptr);    
+ std::shared_ptr<T const> get_shared() const {
+  if (auto ptr = mpark::get_if<shared_t>(&this->ptr)) {
```

Comment:
Why change to returning `nullptr`? These are really only intended to be called when it has already been checked that they have the value (similar to `optional`)

Reply:
OK, I will fix it